### PR TITLE
Fix payment method section missing for Affirm and Afterpay on transaction details page

### LINKED ISSUE #7343: Missing payment details section in payment details page for Affirm and Afterpay payment methods
### Describe the bug
For Affirm and Afterpay/Clearpay payment methods we are missing the "payment method" section in transaction details page. 

### To Reproduce

1. Make a payment with Affirm or Afterpay/Clearpay payment method
2. Go to `payments`->`transactions` and locate latest transaction. Click it to open payment details page.
3. Observe that there is no "payment method" section.
4. Perform steps 1-2. for any other PM like credit card payment. Observe that "payment method" section is available in details page.

### Screenshots
For CC there's "payment method" section
image

For Affirm there's no "payment method" section
image

### Additional context
See `client/payment-details/payment-method/index.js` we have 
```JS

import CardDetails from './card';
import CardPresentDetails from './card-present';
import BancontactDetails from './bancontact';
import BecsDetails from './becs';
import EpsDetails from './eps';
import GiropayDetails from './giropay';
import IdealDetails from './ideal';
import P24Details from './p24';
import SepaDetails from './sepa';
import SofortDetails from './sofort';
import KlarnaDetails from 'wcpay/payment-details/payment-method/klarna';

const detailsComponentMap = {
	card: CardDetails,
	card_present: CardPresentDetails,
	au_becs_debit: BecsDetails,
	bancontact: BancontactDetails,
	eps: EpsDetails,
	giropay: GiropayDetails,
	ideal: IdealDetails,
	p24: P24Details,
	sepa_debit: SepaDetails,
	sofort: SofortDetails,
	klarna: KlarnaDetails,
};
```
which is missing these 2 PMs

### PR DESCRIPTION
**Pull request by mgascam**
Fixes #7343 

#### Changes proposed in this Pull Request

This pull request adds the payment method section to the details page of a transaction paid with Affirm or Afterpay.

### Affirm

**Before**

Screenshot 2023-10-04 at 16 25 24

**After**

Screenshot 2023-10-04 at 16 22 46

### Afterpay

**Before**

Screenshot 2023-10-04 at 16 26 33

**After**

Screenshot 2023-10-04 at 16 25 51

#### Testing instructions

* You'll need at least one Affirm and Afterpay transactio.
* Go to WP Admin > Payments > Transactions
* Locate one Affirm transaction and click it to go the details page
* Scroll to bottom and check that the payment method section appears.
* Repeat the process with one Afterpay transaction
-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) :  'QA Testing Not Applicable'
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7345` or branch name `fix/7343-missing-payment-details-section-in-payment-details-page-for-affirm-and-afterpay-payment-methods` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7343-missing-payment-details-section-in-payment-details-page-for-affirm-and-afterpay-payment-methods) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: bb32d39d4bdd670de490df590bc71e3e091954a2
- Build time: 2023-10-05 11:56:52 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-details/payment-method/index.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ import IdealDetails from './ideal';
 import P24Details from './p24';
 import SepaDetails from './sepa';
 import SofortDetails from './sofort';
+import AffirmDetails from './affirm';
+import AfterpayClearpayDetails from './afterpay-clearpay';
```
Comment (dmvrtx) [inline client/payment-details/payment-method/index.js]: If we sort imports here we will avoid conflicts with #7282

**Review by dmvrtx — APPROVED** [review]
Thank you for addressing this @mgascam. Provisionally approved with a minor comment. Tested on my local environment, looks good.

File: client/payment-details/payment-method/index.js (hunk 1 above)
Comment (mgascam) [reply client/payment-details/payment-method/index.js]: Done here https://github.com/Automattic/woocommerce-payments/pull/7345/commits/07c70fac9f49ea458d328ddb925816251f2f49a9 thanks!

**Review by kalessil — COMMENTED** [review]
Thanks for adding screenshots @mgascam !

Some remarks:
- I'd expect the new components to be in TS (this can be done in a separate PR if this one is aiming before code freeze)
- It'd be better to have the same set of fields (in Affirm case it's less, so we could follow the same approach)
- I'm a bit confused as don't understand where do we invoke the new components

**Comment by kalessil** [comment]
My comments are not show-stoppers but subjects to consider for us and learn something new for me.

**Comment by mgascam** [comment]
Thanks @kalessil , answering your questions below:

> * I'd expect the new components to be in TS (this can be done in a separate PR if this one is aiming before code freeze)

Yes,  it would be great to include this change in next release so I'll give it a try (timeboxed) to perform the migration in this PR. If I don't progress quick enough, I'll tackle it in different PR.

> * It'd be better to have the same set of fields (in Affirm case it's less, so we could follow the same approach)

Indeed, I added the additional fields for Afterpay because Stripe docs [list](https://stripe.com/docs/api/charges/object#charge_object-payment_method_details-afterpay_clearpay) those as possible fields. But truth is those fields don't add much information (we already have the order in this view) so I removed them in this commit https://github.com/Automattic/woocommerce-payments/pull/7345/commits/8240ccae1bcda0a713ee76d5b56995a4d57ef1f3

> * I'm a bit confused as don't understand where do we invoke the new components

The `PaymentDetailsPaymentMethod` is the one that renders the components. It holds a dictionary of all available `PaymentMethodDetails`. It looks up the type informed in the `payment_method_details` of the `charge` in the dictionary and renders it or returns null. It happens here: https://github.com/Automattic/woocommerce-payments/blob/07c70fac9f49ea458d328ddb925816251f2f49a9/client/payment-details/payment-method/index.js#L50

**Review by kalessil — APPROVED** [review]

**Comment by mgascam** [comment]
Sent to the merge queue. 

I created this https://github.com/Automattic/woocommerce-payments/issues/7388 to address the https://github.com/Automattic/woocommerce-payments/pull/7345#issuecomment-1747375749 to TSX in a different PR.
 
FYI: @kalessil

**Pull request merged**